drivers/ethernet/eth_stm32_hal.c

```diff
@@ -32,19 +33,60 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
 #endif
 
+#define PHY_ADDR	CONFIG_ETH_STM32_HAL_PHY_ADDRESS
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+
+#define PHY_BSR  ((uint16_t)0x0001U)  /*!< Transceiver Basic Status Register */
+#define PHY_LINKED_STATUS  ((uint16_t)0x0004U)  /*!< Valid link established */
+
+#define GET_FIRST_DMA_TX_DESC(heth)	(heth->Init.TxDesc)
+#define IS_ETH_DMATXDESC_OWN(dma_tx_desc)	(dma_tx_desc->DESC3 & \
+							ETH_DMATXNDESCRF_OWN)
+
+#define ETH_RXBUFNB	ETH_RX_DESC_CNT
+#define ETH_TXBUFNB	ETH_TX_DESC_CNT
+
+#define ETH_MEDIA_INTERFACE_MII		HAL_ETH_MII_MODE
+#define ETH_MEDIA_INTERFACE_RMII	HAL_ETH_RMII_MODE
+
+#define LL_AHB1_GRP1_PERIPH_ETHMAC	LL_AHB1_GRP1_PERIPH_ETH1MAC
+#define LL_AHB1_GRP1_PERIPH_ETHMACTX	LL_AHB1_GRP1_PERIPH_ETH1TX
+#define LL_AHB1_GRP1_PERIPH_ETHMACRX	LL_AHB1_GRP1_PERIPH_ETH1RX
+#define LL_AHB1_GRP1_PERIPH_ETHMACPTP	LL_AHB1_GRP1_PERIPH_ETH1MAC
+
+#define ETH_DMA_TRANSMIT_TIMEOUT	20U  /* transmit timeout in sys tiks */
+
+#else
+
+#define GET_FIRST_DMA_TX_DESC(heth)	(heth->TxDesc)
+#define IS_ETH_DMATXDESC_OWN(dma_tx_desc)	(dma_tx_desc->Status & \
+							ETH_DMATXDESC_OWN)
+
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+
 #if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
 	    DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
-static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __dtcm_noinit_section;
-static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __dtcm_noinit_section;
-static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __dtcm_noinit_section;
-static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] __dtcm_noinit_section;
+#define ETH_DMA_MEM	__dtcm_noinit_section
 #else
-static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __aligned(4);
-static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __aligned(4);
-static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __aligned(4);
-static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] __aligned(4);
+#define ETH_DMA_MEM	__aligned(4)
 #endif /* CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER */
 
+#if defined(CONFIG_NOCACHE_MEMORY)
+#define NO_CACHE __nocache
+#else
+#define NO_CACHE
+#endif  /* CONFIG_NOCACHE_MEMORY */
+
+static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] NO_CACHE ETH_DMA_MEM;
+static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] NO_CACHE ETH_DMA_MEM;
+static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] NO_CACHE ETH_DMA_MEM;
+static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] NO_CACHE ETH_DMA_MEM;
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+static ETH_TxPacketConfig tx_config NO_CACHE;
+#endif
```

Comment:
Just a comment on this one:
I wonder if there could be an alternative to `NO_CACHE` so that one doesn't conclude to fast that these symbols are not cached in any case

Reply:
1. According to [CONFIG_NOCACHE_MEMORY](https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_NOCACHE_MEMORY.html) `NOCACHE_MEMORY` just allocates non-cached memory area.

2. Taking 1 into account the corresponding memory buffers will be placed in this `NOCACHE_MEMORY` if it exist other way not. Thus exactly `nocache` functionality will be used here and `NO_CACHE` is okay I think.

3. I would just add following comment on `NO_CAHE` define: 
`/*  On some STM32 (e.g. H7 series) the non-cached memory needed to avoid ETH DMA data inconsistency. The NO_CACHE lets linker place corresponding buffers into non-cached memory area which must be previously created by CONFIG_NOCACHE_MEMORY KConfig definition */`

Are you okay with it?

Reply:
Im fine with the functionality, my comment was purely on the name of this macro, as `NO_CACHE` could actually mean 'cached' in some cases. I'd like to avoid potential confusions if possible

Reply:
You're right about naming. Especially on the line 78 it can be clearly seen: `NO_CACHE` used, but it is still cached.
**How about `CACHE` name of this define?** Then it does not says about certain usage of the memory (cached/non-cached) but still means memory type which can be some kind of cache her in context.

Reply:
Go for `CACHE`

Reply:
Done